WordPress/Classes/ViewRelated/System/WPSplitViewController.swift

```diff
@@ -0,0 +1,163 @@
+    }
+
+    override func overrideTraitCollectionForChildViewController(childViewController: UIViewController) -> UITraitCollection? {
+        guard let collection = super.overrideTraitCollectionForChildViewController(childViewController) else { return nil }
+
+        let overrideCollection = UITraitCollection(horizontalSizeClass: self.traitCollection.horizontalSizeClass)
+        return UITraitCollection(traitsFromCollections: [collection, overrideCollection])
+    }
+
+    override var viewControllers: [UIViewController] {
+        didSet {
+            // Ensure that each top level navigation controller has
+            // `extendedLayoutIncludesOpaqueBars` set to true. Otherwise we
+            // see a large tab bar sized gap underneath each view controller.
+            for viewController in viewControllers {
+                if let viewController = viewController as? UINavigationController {
+                    viewController.extendedLayoutIncludesOpaqueBars = true
+                }
+            }
+        }
+    }
+
+    override func showDetailViewController(vc: UIViewController, sender: AnyObject?) {
+        var detailVC = vc
+
+        // Ensure that detail view controllers are wrapped in a navigation controller
+        if !(vc is UINavigationController || collapsed) {
+            detailVC = UINavigationController(rootViewController: vc)
+        }
+
+        super.showDetailViewController(detailVC, sender: self)
+    }
+
+    /** Sets the primary view controller of the split view as specified, and
+     *  automatically sets the detail view controller if the primary
+     *  conforms to `WPSplitViewControllerDetailProvider` and can vend a
+     *  detail view controller.
+     */
+    func setInitialPrimaryViewController(viewController: UIViewController) {
+        var initialViewControllers = [viewController]
+
+        if let navigationController = viewController as? UINavigationController,
+            let rootViewController = navigationController.viewControllers.first,
+            let detailViewController = initialDetailViewControllerForPrimaryViewController(rootViewController) {
+
+            navigationController.delegate = self
+
+            initialViewControllers.append(detailViewController)
+
+            viewControllers = initialViewControllers
+        }
+    }
+
+    private func initialDetailViewControllerForPrimaryViewController(viewController: UIViewController) -> UIViewController? {
+        guard let detailProvider = viewController as? WPSplitViewControllerDetailProvider,
+        let detailViewController = detailProvider.initialDetailViewControllerForSplitView(self)  else {
+            return nil
+        }
+
+        // Ensure it's wrapped in a navigation controller
+        if detailViewController is UINavigationController {
+            return detailViewController
+        } else {
+            return UINavigationController(rootViewController: detailViewController)
+        }
+    }
```

Comment:
Maybe both `initialDetailViewControllerForPrimaryViewController`  and `showDetailViewController` should share the wrapping of navigation controllers, to avoid maintaining this particular step in two spots, but it is pretty simple. 🤔